mmdet/core/evaluation/eval_hooks.py

```diff
@@ -173,16 +173,16 @@ def save_best_checkpoint(self, runner, key_score):
                              f'Best {self.key_indicator} is {best_score:0.4f}')
 
     def evaluate(self, runner, results):
-        eval_res = self.dataloader.dataset.evaluate(
+        self.eval_res = self.dataloader.dataset.evaluate(
```

Comment:
Agreed, I've switched to a workaround using the Runner's `log_buffer` attribute. I've modified `EvalHook` in the first place to ensure that all of the derived hooks like `DistEvalHook` had an attribute containing the computed metric values. Otherwise I would need to introduce several custom EvalHooks.